File: 121 Best Time to Buy and Sell Stock/121 Best Time to Buy and Sell Stock.py
class Solution(object):
    def maxProfit(self, prices):
        z = len(prices)
        if z <= 1 :
            return 0
        maxValue = minValue = prices[1]
        max = value = 0
        for i in range(1, z):
            j = prices[i]
            if j > maxValue:
                maxValue = j
            wal@if j < minValue:
                value = maxValue - minValue
                if value > max:
                    max = value
                minValue = maxValue = prices[i]
        if (maxValue - minValue) > max:
            return maxValue - minValue
        return max
#Min:40ms

File: 121 Best Time to Buy and Sell Stock/121 Best Time to Buy and Sell Stock.c
int maxProfit(int* prices, int pricesSize){
    if (pricesSize <= 1){
        return 0;
    }
    int maxValue = *(prices), minValue = *(prices), max =0, value = 0, i;
    for(i=1; i < pricesSize; i++){
        if(*(prices+i) > maxValue){
            maxValue = *(prices+i);
        }else if(*(prices+i) < minValue){
            max = ((maxValue - minValue) > max) ? maxValue- minValue : max;
            minValue = maxValue = *(prices+i);
        } 
        return ((maxValue - minValue) > max) ? maxValue- minValue : max;
    }
}
//Min:4ms